dialogue-core/src/test/java/com/palantir/dialogue/core/RetryingChannelTest.java

```diff
@@ -531,15 +536,62 @@ public void retriesSocketTimeout_connectionTimeout() throws ExecutionException,
     }
 
     @Test
-    public void nonRetryableRequestBodyIsNotRetried() throws ExecutionException, InterruptedException {
-        when(channel.execute(any())).thenReturn(FAILED).thenReturn(SUCCESS);
+    public void requestWithNonRepeatableBodyRetriedWhenConnectionFails()
```

Comment:
Can we rearrange the change to the tests for a cleaner diff? I'd prefer if we could make the minimal possible change to the existing test such that it still asserts the same thing, and add new test cases to cover new considerations. It makes reviewing much easier :-)